### ISSUE TITLE
Default pool size for the Redis RedisMessageListenerContainer.taskExecutor

### ISSUE THREAD

**Issue by rolandgroen** [spring-projects/spring-session#485]
The RedisMessageListenerContainer uses a SimpleAsyncTaskExecutor to dispatch the messages. This causes problems regarding thread creation and repletion, see issue https://github.com/spring-projects/spring-session/issues/483. 

It makes sense to make spring-session inject a Pooled TaskExecutor into spring-data-redis. We use a pool size of 10.

**Comment by rwinch** [comment]
@rolandgroen Thanks!

cc I believe you were interested in this too @tsachev

**Comment by tsachev** [comment]
@rwinch Yes I opened #314 some time ago and after a discussion we agreed to leave `SimpleAsyncTaskExecutor` as default and provide an extension point for the executors. This was implemented in PR #374.
I agree that `SimpleAsyncTaskExecutor` is terrible default in case of `spring-session`.
In my project I think we use something like the one configured in this test
https://github.com/spring-projects/spring-session/blob/master/spring-session/src/integration-test/java/org/springframework/session/data/redis/taskexecutor/RedisListenerContainerTaskExecutorITests.java#L100

I.e. single thread executor for events and the `SimpleAsyncTaskExecutor` for the subscription.

**Comment by kunalvarpe** [comment]
I would like contribute, Can someone mentor/help me here?

**Comment by lsh1215** [comment]
Hi, I’m thinking of taking a look at this one.

From a quick check, it seems like #314 / #374 added a way to plug in custom Redis listener executors, but the default still falls back to `SimpleAsyncTaskExecutor` when no custom beans are defined.

Is this still something you’d like to change? If so, is there anything I should keep in mind before I start on a PR?

**Comment by lsh1215** [comment]
Following up on my earlier question, I looked a bit deeper into the execution model here.

It seems like treating this only as a “default pool size” change may be a bit too narrow. If Spring Session only sets a default `RedisMessageListenerContainer#setTaskExecutor`, Spring Data Redis will use that same executor as the `subscriptionExecutor` when no separate subscription executor is configured. Since subscription work is long-running while message dispatch is bursty, that could accidentally couple two different execution concerns.

So the safer direction seems to be:

- keep honoring user-provided `springSessionRedisTaskExecutor`
- keep honoring user-provided `springSessionRedisSubscriptionExecutor`
- provide a safer default for message dispatch so the default path does not create an unbounded number of platform threads
- avoid accidentally using that dispatch executor for Redis subscription work
- make lifecycle/shutdown ownership explicit for any executor Spring Session creates

Before I work on a PR, does this framing match the intent of this issue? Or would you prefer this to stay scoped to a simpler `taskExecutor` default / pool-size change?

If this direction sounds reasonable, I can prepare a small draft PR with tests covering the default case and the existing task/subscription executor override cases.
